crates/common/src/opa_executor.rs

```diff
@@ -297,9 +328,29 @@ impl OpaExecutor for WasmtimeOpaExecutor {
         context: &serde_json::Value,
     ) -> Result<bool, OpaExecutorError> {
         self.opa.set_data(context)?;
-        let input = IdentityContext::new(id);
-        let res = self.opa.eval(&self.entrypoint, &input)?;
-        Ok(res)
+        let input = id.identity_context()?;
+        match self.opa.eval(&self.entrypoint, &input)? {
+            true => Ok(true),
+            false => Err(OpaExecutorError::AccessDenied { id: id.to_string() }),
+        }
+    }
+}
+
+pub trait EvaluationContext {
+    /// Transform the @graph of an operations dependencies in a ProvModel
+    /// into a more useful representation for looking up from an operation
+    fn format_context_data(&mut self) -> Result<serde_json::Value, ProcessorError>;
+}
+
+impl EvaluationContext for serde_json::Value {
```

Comment:
Hmm, I am not sure I like this being on Value, can we use a newtype wrapper or some sort that indicates it's a serialized / compacted ProvModel

Reply:
I scrapped the trait. At first I thought I could use a newtype but implementing derefmut, according to the docs, should only be for smart pointers.